task-docs/init/task.md

```diff
@@ -0,0 +1,236 @@
+# `init/boilerplate`
+
+```
+{
+	interactive: true,
+	flags: [
+		--migration ?
+		--fixtures ?
+	],
+	functionality: [
+		'component boilerplate',
+		'demo (app?) boilerplate'
+		'test file boilerplate',
+		'fixture boilerplate'
+	],
+	requires: [
+		'a set up to support a react app?'
+	]
+}
+```
+
+This task should be responsible for a more comprehensive boilerplate than the current `obt init`.
+
+We can provide more granular control over what boilerplate details get output by making this command interactive, much like an `npm init`—meaning we need a list of prompts for the command.
```

Comment:
i love this a lot. it would be cool if you could do `oat init js` if you're later adding javascript to a component that didn't used to have one.

like `rails generate` do, i think.

though it's possible that would be sticking two things into one command and they should be handled separately by separate `oat new` and `oat gen` commands

Reply:
We had discussed using flags, e.g. `oat generate [app]` (for all things) or something like `oat generate --js` but we decided against it and I can't remember why now help @JakeChampion

Reply:
or rather, if the file doesn't already exist in the tree, then you get a prompt for it?
like 👇 line 40